### PR TITLE
Support updates and upserts with blank dates represented by strings

### PR DESCRIPTION
**Pull request by prescod**
A dataset like this:

```
BEGIN TRANSACTION;
CREATE TABLE "Contact" (
	id INTEGER NOT NULL, 
	"FirstName" VARCHAR(255), 
	"LastName" VARCHAR(255), 
	"Birthdate" VARCHAR(255), 
	"External_ID__c" VARCHAR(255), 
	"IsPersonAccount" VARCHAR(255), 
	PRIMARY KEY (id)
);
INSERT INTO "Contact" VALUES(1,'AAA','BBBB','','C2','False');
COMMIT;
```

Previously generated an error like this:

```
                    WARNING  Error on record with id 1: JSON_PARSER_ERROR: Cannot deserialize   utils.py:141
                             instance of date from VALUE_STRING value  or request may be                    
                             missing a required field at [Source: N/A; line: -1, column:                    
                             -1][line:-1, column:-1] ()                                                     
```

### CODE REVIEW

**Comment by prescod** [comment]
Fixes #3283

File: cumulusci/tasks/bulkdata/tests/test_step.py [hunk 1]
```diff
@@ -1386,3 +1386,66 @@ def test_get_dml_operation__bad_api(self, rest_dml, bulk_dml):
...
+            "fields": [
+                {"name": "Birthdate", "type": "date"},
+                {"name": "IsHappy", "type": "boolean"},
+                {"name": "Name", "type": "string"},
+            ],
+        }
+
+        step = get_dml_operation(
+            sobject="Test__c",
+            operation=DataOperationType.UPSERT,
```
Comment (boakley) [inline cumulusci/tasks/bulkdata/tests/test_step.py:1436]: `_record_to_json` has a branch that includes both UPDATE and UPSERT but this test only covers UPSERT. Should there be a test for UPDATE?

**Review by boakley — APPROVED** [review]
This looks pretty good to me. I made one small comment, but it's not a show-stopper.

**Review by boakley — APPROVED** [review]
Looks good to me. I've never used `@pytest.mark.parameterize` before, now I have a reason to read up on it.

**Pull request merged**